R/tm_g_ae_oview.R

```diff
@@ -285,34 +284,39 @@ srv_g_ae_oview <- function(id,
 
     output_q <- reactive({
       ANL <- data[[dataname]]() # nolint
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
-      validate(need(
-        is.factor(ANL[[input$arm_var]]),
-        "Selected arm variable needs to be a factor."
+
+      teal::validate_has_data(ANL, min_nrow = 10, msg = sprintf("%s has not enough data", dataname))
+
+      iv <- shinyvalidate::InputValidator$new()
```

Comment:
What do you think about moving the initialization of `iv` and its rules in a separate reactive at the top of the module and calling `teal::validate_inputs` only in `output_q`?
 This would be similar to what is introduced in tmc https://github.com/insightsengineering/teal.modules.clinical/pull/699/files and looks cleaner in a way that the validation are separate from the output call.